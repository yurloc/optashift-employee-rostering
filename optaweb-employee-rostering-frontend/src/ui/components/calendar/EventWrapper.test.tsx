/*
 * Copyright 2019 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import * as React from 'react';
import { shallow } from 'enzyme';
import EventWrapper from './EventWrapper';

describe('EventWrapper', () => {
  it('should render correctly when the event does not span multiple days', () => {
    const eventWrapper = shallow(
      <EventWrapper
        continuesEarlier={false}
        continuesLater={false}
        className="class"
        popoverHeader="Title"
        popoverBody="Body"
        style={{ top: '50%', height: '25%', color: 'white' }}
      />,
    );
    expect(eventWrapper).toMatchSnapshot();
  });

  it('should render correctly when the event continues from eariler', () => {
    const eventWrapper = shallow(
      <EventWrapper
        continuesEarlier
        continuesLater={false}
        className="class"
        popoverHeader="Title"
        popoverBody="Body"
        style={{ top: '0%', height: '25%', color: 'white' }}
      />,
    );
    expect(eventWrapper).toMatchSnapshot();
  });

  it('should render correctly when the event continues later', () => {
    const eventWrapper = shallow(
      <EventWrapper
        continuesEarlier={false}
        continuesLater
        className="class"
        popoverHeader="Title"
        popoverBody="Body"
        style={{ top: '75%', height: '25%', color: 'white' }}
      />,
    );
    expect(eventWrapper).toMatchSnapshot();
  });

  it('should render correctly when the event continues both before and after', () => {
    const eventWrapper = shallow(
      <EventWrapper
        continuesEarlier
        continuesLater
        className="class"
        popoverHeader="Title"
        popoverBody="Body"
        style={{ top: '0%', height: '100%', color: 'white' }}
      />,
    );
    expect(eventWrapper).toMatchSnapshot();
  });

  it('should render correctly when the event has no height or top', () => {
    const eventWrapper = shallow(
      <EventWrapper
        continuesEarlier={false}
        continuesLater={false}
        className="class"
        popoverHeader="Title"
        popoverBody="Body"
        style={{ color: 'white' }}
      />,
    );
    expect(eventWrapper).toMatchSnapshot();
  });

  it('should render correctly without a style', () => {
    const eventWrapper = shallow(
      <EventWrapper
        continuesEarlier={false}
        continuesLater={false}
        className="class"
        popoverHeader="Title"
        popoverBody="Body"
      />,
    );
    expect(eventWrapper).toMatchSnapshot();
  });

  it('should render correctly when the event does not have a class name', () => {
    const eventWrapper = shallow(
      <EventWrapper
        continuesEarlier={false}
        continuesLater={false}
        className=""
        popoverHeader="Title"
        popoverBody="Body"
        style={{ top: '50%', height: '25%', color: 'white' }}
      />,
    );
    expect(eventWrapper).toMatchSnapshot();
  });
});
