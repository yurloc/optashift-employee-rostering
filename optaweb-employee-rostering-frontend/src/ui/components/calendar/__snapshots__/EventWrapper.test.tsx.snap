// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventWrapper should render correctly when the event continues both before and after 1`] = `
<div
  className="class continues-from-previous-day continues-next-day"
  style={
    Object {
      "backgroundColor": "transparent",
      "border": "none",
      "color": "white",
      "height": "100%",
      "top": "0%",
    }
  }
>
  <Popover
    appendTo={[Function]}
    aria-label=""
    bodyContent="Body"
    boundary="window"
    className=""
    closeBtnAriaLabel="Close"
    distance={25}
    enableFlip={true}
    flipBehavior={
      Array [
        "top",
        "right",
        "bottom",
        "left",
        "top",
        "right",
        "bottom",
      ]
    }
    footerContent={null}
    headerContent="Title"
    hideOnOutsideClick={true}
    isVisible={null}
    maxWidth="calc(2rem + 18.75rem)"
    onHidden={[Function]}
    onHide={[Function]}
    onMount={[Function]}
    onShow={[Function]}
    onShown={[Function]}
    position="right"
    shouldClose={[Function]}
    zIndex={9999}
  >
    <div />
  </Popover>
</div>
`;

exports[`EventWrapper should render correctly when the event continues from eariler 1`] = `
<div
  className="class continues-from-previous-day"
  style={
    Object {
      "backgroundColor": "transparent",
      "border": "none",
      "color": "white",
      "height": "25%",
      "top": "0%",
    }
  }
>
  <Popover
    appendTo={[Function]}
    aria-label=""
    bodyContent="Body"
    boundary="window"
    className=""
    closeBtnAriaLabel="Close"
    distance={25}
    enableFlip={true}
    flipBehavior={
      Array [
        "top",
        "right",
        "bottom",
        "left",
        "top",
        "right",
        "bottom",
      ]
    }
    footerContent={null}
    headerContent="Title"
    hideOnOutsideClick={true}
    isVisible={null}
    maxWidth="calc(2rem + 18.75rem)"
    onHidden={[Function]}
    onHide={[Function]}
    onMount={[Function]}
    onShow={[Function]}
    onShown={[Function]}
    position="right"
    shouldClose={[Function]}
    zIndex={9999}
  >
    <div />
  </Popover>
</div>
`;

exports[`EventWrapper should render correctly when the event continues later 1`] = `
<div
  className="class continues-next-day"
  style={
    Object {
      "backgroundColor": "transparent",
      "border": "none",
      "color": "white",
      "height": "25%",
      "top": "75%",
    }
  }
>
  <Popover
    appendTo={[Function]}
    aria-label=""
    bodyContent="Body"
    boundary="window"
    className=""
    closeBtnAriaLabel="Close"
    distance={25}
    enableFlip={true}
    flipBehavior={
      Array [
        "top",
        "right",
        "bottom",
        "left",
        "top",
        "right",
        "bottom",
      ]
    }
    footerContent={null}
    headerContent="Title"
    hideOnOutsideClick={true}
    isVisible={null}
    maxWidth="calc(2rem + 18.75rem)"
    onHidden={[Function]}
    onHide={[Function]}
    onMount={[Function]}
    onShow={[Function]}
    onShown={[Function]}
    position="right"
    shouldClose={[Function]}
    zIndex={9999}
  >
    <div />
  </Popover>
</div>
`;

exports[`EventWrapper should render correctly when the event does not have a class name 1`] = `
<div
  className=""
  style={
    Object {
      "backgroundColor": "transparent",
      "border": "none",
      "color": "white",
      "height": "25%",
      "top": "50%",
    }
  }
>
  <Popover
    appendTo={[Function]}
    aria-label=""
    bodyContent="Body"
    boundary="window"
    className=""
    closeBtnAriaLabel="Close"
    distance={25}
    enableFlip={true}
    flipBehavior={
      Array [
        "top",
        "right",
        "bottom",
        "left",
        "top",
        "right",
        "bottom",
      ]
    }
    footerContent={null}
    headerContent="Title"
    hideOnOutsideClick={true}
    isVisible={null}
    maxWidth="calc(2rem + 18.75rem)"
    onHidden={[Function]}
    onHide={[Function]}
    onMount={[Function]}
    onShow={[Function]}
    onShown={[Function]}
    position="right"
    shouldClose={[Function]}
    zIndex={9999}
  >
    <div />
  </Popover>
</div>
`;

exports[`EventWrapper should render correctly when the event does not span multiple days 1`] = `
<div
  className="class"
  style={
    Object {
      "backgroundColor": "transparent",
      "border": "none",
      "color": "white",
      "height": "25%",
      "top": "50%",
    }
  }
>
  <Popover
    appendTo={[Function]}
    aria-label=""
    bodyContent="Body"
    boundary="window"
    className=""
    closeBtnAriaLabel="Close"
    distance={25}
    enableFlip={true}
    flipBehavior={
      Array [
        "top",
        "right",
        "bottom",
        "left",
        "top",
        "right",
        "bottom",
      ]
    }
    footerContent={null}
    headerContent="Title"
    hideOnOutsideClick={true}
    isVisible={null}
    maxWidth="calc(2rem + 18.75rem)"
    onHidden={[Function]}
    onHide={[Function]}
    onMount={[Function]}
    onShow={[Function]}
    onShown={[Function]}
    position="right"
    shouldClose={[Function]}
    zIndex={9999}
  >
    <div />
  </Popover>
</div>
`;

exports[`EventWrapper should render correctly when the event has no height or top 1`] = `
<div
  className="class"
  style={
    Object {
      "backgroundColor": "transparent",
      "border": "none",
      "color": "white",
    }
  }
>
  <Popover
    appendTo={[Function]}
    aria-label=""
    bodyContent="Body"
    boundary="window"
    className=""
    closeBtnAriaLabel="Close"
    distance={25}
    enableFlip={true}
    flipBehavior={
      Array [
        "top",
        "right",
        "bottom",
        "left",
        "top",
        "right",
        "bottom",
      ]
    }
    footerContent={null}
    headerContent="Title"
    hideOnOutsideClick={true}
    isVisible={null}
    maxWidth="calc(2rem + 18.75rem)"
    onHidden={[Function]}
    onHide={[Function]}
    onMount={[Function]}
    onShow={[Function]}
    onShown={[Function]}
    position="right"
    shouldClose={[Function]}
    zIndex={9999}
  >
    <div />
  </Popover>
</div>
`;

exports[`EventWrapper should render correctly without a style 1`] = `
<div
  className="class"
  style={
    Object {
      "backgroundColor": "transparent",
      "border": "none",
    }
  }
>
  <Popover
    appendTo={[Function]}
    aria-label=""
    bodyContent="Body"
    boundary="window"
    className=""
    closeBtnAriaLabel="Close"
    distance={25}
    enableFlip={true}
    flipBehavior={
      Array [
        "top",
        "right",
        "bottom",
        "left",
        "top",
        "right",
        "bottom",
      ]
    }
    footerContent={null}
    headerContent="Title"
    hideOnOutsideClick={true}
    isVisible={null}
    maxWidth="calc(2rem + 18.75rem)"
    onHidden={[Function]}
    onHide={[Function]}
    onMount={[Function]}
    onShow={[Function]}
    onShown={[Function]}
    position="right"
    shouldClose={[Function]}
    zIndex={9999}
  >
    <div />
  </Popover>
</div>
`;
